### PR TITLE
fix order in DataTable with entity containing embbed

### PR DESCRIPTION
**Pull request by veolys-nanne**

### CODE REVIEW

File: Pagination/MongoTrait/FilterTrait.php [hunk 1]
```diff
@@ -138,22 +139,42 @@ protected function generateFilterSearchField($name, $value, $type)
     }
 
     /**
-     * @param Stage       $qa
      * @param array|null  $order
      * @param array|null  $descriptionEntity
      *
      * @return Stage
```
Comment (nicolasThal) [inline Pagination/MongoTrait/FilterTrait.php]: the method does not return a `Stage` anymore

File: Pagination/MongoTrait/FilterTrait.php [hunk 2]
```diff
@@ -138,22 +139,42 @@ protected function generateFilterSearchField($name, $value, $type)
...
-                $name = $descriptionEntity[$columnsName]['field'];
-                $dir = ($order['dir'] == 'desc') ? -1 : 1;
-                $qa->sort(array($name => $dir));
+            if (isset($descriptionEntity[$columnsName])){
+                if($descriptionEntity[$columnsName] instanceof Search) {
+                    $name = $descriptionEntity[$columnsName]->getField();
+                } elseif (isset($descriptionEntity[$columnsName]['field'])) {
+                    $name = $descriptionEntity[$columnsName]['field'];
+                }
+                $dir = $name;
```
Comment (nicolasThal) [inline Pagination/MongoTrait/FilterTrait.php]: dir is usually an alias for directory. what dos it stand for ?

File: Pagination/MongoTrait/FilterTrait.php [hunk 3]
```diff
@@ -138,22 +139,42 @@ protected function generateFilterSearchField($name, $value, $type)
...
+                if ($returnOrder) {
+                    $dir = ($order['dir'] == 'desc') ? -1 : 1;
+                }
+                return array(str_replace('.', '_', $name) => $dir);
```
Comment (nicolasThal) [inline Pagination/MongoTrait/FilterTrait.php]: empty line before return

File: Pagination/MongoTrait/FilterTrait.php [hunk 4]
```diff
@@ -138,22 +139,42 @@ protected function generateFilterSearchField($name, $value, $type)
...
-                $dir = ($order['dir'] == 'desc') ? -1 : 1;
-                $qa->sort(array($name => $dir));
+            if (isset($descriptionEntity[$columnsName])){
+                if($descriptionEntity[$columnsName] instanceof Search) {
+                    $name = $descriptionEntity[$columnsName]->getField();
+                } elseif (isset($descriptionEntity[$columnsName]['field'])) {
+                    $name = $descriptionEntity[$columnsName]['field'];
+                }
+                $dir = $name;
+                if ($returnOrder) {
```
Comment (itkg-alavieille) [inline Pagination/MongoTrait/FilterTrait.php]: why add a condition ?

File: Pagination/MongoTrait/FilterTrait.php (hunk 2 above)
Comment (alavieille) [inline Pagination/MongoTrait/FilterTrait.php]: in the case where $descriptionEntity[$columnsName] isn't an instance of Search and the key field not exists the parameter $name doesn't exist

File: Pagination/MongoTrait/FilterTrait.php [hunk 5]
```diff
@@ -138,22 +139,42 @@ protected function generateFilterSearchField($name, $value, $type)
...
+    protected function generateArrayForFilterSort($order = null , $descriptionEntity = null, $returnOrder = true)
     {
         if (null !== $order) {
             $columnsName = $order['name'];
-            if (isset($descriptionEntity[$columnsName]) && isset($descriptionEntity[$columnsName]['field'])) {
-                $name = $descriptionEntity[$columnsName]['field'];
-                $dir = ($order['dir'] == 'desc') ? -1 : 1;
-                $qa->sort(array($name => $dir));
+            if (isset($descriptionEntity[$columnsName])){
+                if($descriptionEntity[$columnsName] instanceof Search) {
```
Comment (ngilain) [inline Pagination/MongoTrait/FilterTrait.php]: if (

File: Pagination/MongoTrait/FilterTrait.php (hunk 2 above)
Comment (veolys-nanne) [inline Pagination/MongoTrait/FilterTrait.php]: ^^, It was for direction, not directory. Will replace it.

**Comment by veolys-nanne** [comment]
done @itkg-ngilain , @nicolasThal , @itkg-alavieille

File: Pagination/MongoTrait/FilterTrait.php [hunk 6]
```diff
@@ -138,23 +140,69 @@ protected function generateFilterSearchField($name, $value, $type)
     }
 
     /**
-     * @param Stage       $qa
+     * @param array|null  $order
```
Comment (alavieille) [inline Pagination/MongoTrait/FilterTrait.php]: update comment parameter by PaginateFinderConfiguration $configuration

File: Pagination/MongoTrait/FilterTrait.php [hunk 7]
```diff
@@ -138,23 +140,69 @@ protected function generateFilterSearchField($name, $value, $type)
...
+    protected function generateGroupForFilterSort(PaginateFinderConfiguration $configuration)
+    {
+        $group = array();
+
+        $sorts = $this->generateArrayForFilterSort(
+            $configuration->getOrder(),
+            $configuration->getDescriptionEntity(),
+            false
+        );
+        foreach($sorts as $key => $name) {
```
Comment (alavieille) [inline Pagination/MongoTrait/FilterTrait.php]: foreach ($sorts as $key => $name)

**Comment by veolys-nanne** [comment]
done @itkg-ngilain , @nicolasThal , @itkg-alavieille

**Comment by alavieille** [comment]
:+1:

**Pull request merged**